tests/api_validation_test.go

```diff
@@ -0,0 +1,160 @@
+package tests
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	dataVolumeName     = "test-dv"
+	validURL           = "http://www.example.com/example.img"
+	invalidURLFormat   = "invalidURL"
+	datavolumeTestFile = "tests/manifests/datavolume.yaml"
+	destinationFile    = "/var/tmp/datavolume_test.yaml"
+)
+
+var _ = Describe("Validation tests", func() {
+
+	f := framework.NewFrameworkOrDie("api-validation-func-test")
+
+	Describe("Verify DataVolume validation", func() {
+		Context("when creating Datavolume", func() {
+			dv := map[string]interface{}{}
+
+			AfterEach(func() {
+				err := os.Remove(destinationFile)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			table.DescribeTable("with Datavolume source validation should", func(sourceType string, args ...string) {
+
+				err := yamlFiletoStruct(datavolumeTestFile, &dv)
+				Expect(err).ToNot(HaveOccurred())
+
+				switch sourceType {
+				case "http":
+					url := args[0]
+					dv["spec"].(map[string]interface{})["source"] = map[string]interface{}{"http": map[string]interface{}{"url": url}}
+
+				case "s3":
+					url := args[0]
+					dv["spec"].(map[string]interface{})["source"] = map[string]interface{}{"s3": map[string]interface{}{"url": url}}
+				case "pvc":
+					namespace := args[0]
+					name := args[1]
+					dv["spec"].(map[string]interface{})["source"] = map[string]interface{}{
+						"pvc": map[string]interface{}{
+							"namespace": namespace,
+							"name":      name}}
+				}
+
```

Comment:
Don't you want to check all other sources (none, registry, blank, upload)?

Reply:
I added the tests according to the test plan.

Reply:
We can merge this and then update the test case and this code.